git-srep/src/main.rs

```diff
@@ -0,0 +1,176 @@
+use clap::Parser;
+use colored::Colorize;
+use git_utils_shared as git_utils;
+use rayon::prelude::*;
+use std::process::Command;
+
+/// Replace given phrase inside files of the provided paths.
+#[derive(Parser, Debug)]
+#[clap(author, version, about, long_about = None)]
+struct Args {
+    /// The word to substitute
+    search: String,
+    /// The word to substitute with
+    replace: String,
+
+    #[clap(
+        long,
+        short,
+        takes_value = false,
+        help = "Substituted word in git dirty files"
+    )]
+    force: bool,
+
+    #[clap(required = true)]
+    paths: Vec<String>,
+}
+
+fn is_file_dirty(file_path: &str) -> bool {
+    let modified = git_utils::file::file_is_modified(file_path);
+    let untracked = git_utils::file::file_is_untracked(file_path);
+    if modified {
+        println!("modified -> {}", file_path.red());
+    }
+    if untracked {
+        println!("Untracked -> {}", file_path.red())
+    }
+
+    modified || untracked
+}
+
+fn change_word_in_files(file_path: Vec<String>, search: &str, replace: &str) {
+    let _ = Command::new("sed")
+        .args([
+            "-i",
+            &format!(
+                "s/{}/{}/g",
+                search.replace("/", r"\/"),
+                replace.replace("/", r"\/")
+            ),
+        ])
+        .args(file_path)
+        .status()
+        .expect("Failed ")
+        .success();
+}
```

Comment:
Make sure when invoking `sed` to use the `-E` option so the regex engine matches that passed to `git grep`